api/client.go

```diff
@@ -40,6 +40,9 @@ type Config struct {
 	// The address of the Prometheus to connect to.
 	Address string
 
+	// Extra headers to add to each request (ex: authorization)
+	Headers map[string]string
```

Comment:
HTTP headers can have multiple values, so this should probably be `map[string][]string`.